Source/WTF/wtf/MallocSpan.h

```diff
@@ -67,33 +68,39 @@ template<typename T, typename Malloc = FastMalloc> class MallocSpan {
 
     static MallocSpan malloc(size_t size)
     {
-        return MallocSpan { static_cast<T*>(Malloc::malloc(size)), size };
+        return MallocSpan { static_cast<T*>(Malloc::malloc(CheckedSize(size) * sizeof(T))), size };
```

Comment:
Well, there's an alias `using CheckedSize = Checked<size_t, RecordOverflow>`, which detects overflow _without_ aborting on overflow (because of the `RecordOverflow`). That's why I suggested `CheckedSize` for the `try[Zeroed]{M,Re}alloc` functions.

Here, for the non-`try*` functions, we should be using `Checked<size_t>`, which is effectively `Checked<size_t, CrashOnOverflow>` because `CrashOnOverflow` is the default template argument value.

Reply:
I still think my code is correct? My code will crash as expected thanks to this logic here:
```
    operator T() const
    {
        if (UNLIKELY(this->hasOverflowed()))
            this->crash();
        return m_value;
    }
```

My CheckedSize gets implicitly converted to a size_t to pass to malloc(), and because it has overflowed, the conversion operator will call crash().

Reply:
Yes, the design of `Checked` is strange, and it’s actually almost never important to use `CrashOnOverflow`, despite how the name sounds. We still end up crashing if there is overflow even without that flag. Might be nice to simplify `Checked` some day to avoid that confusion.

Reply:
I think it’s subtle to change the semantic here without changing names. It’s non-traditional for a function named `malloc` to take a count of objects rather than a count of bytes. Many C++ programmers know that `malloc` takes a byte count and `calloc` and `new` take an object count. I can imagine callers passing a byte count, and then we’d make a too-large allocation, and we might never even notice since it’s wasteful but otherwise harmless.

I suggest we consider changing these function names to not literally contain `malloc` when making this change.

Reply:
Do you prefer function renames to keeping `malloc()` and taking in a `sizeInBytes` instead?

Reply:
I don’t have a strong preference either way.

Reply:
@cdumez Ah, you're right, I managed to miss the `this->crash()` in the conversion operator. I agree with Darin that the `Checked` design is indeed a bit odd, it fooled me a bit today after all!

Reply:
FWIW, I prefer sizeInBytes (as you've done it in your most recent patch). For clients who actually want "malloc" for whatever reason, sizeInBytes is how "malloc" works. For clients who want objects of type T, probably they should use FixedVector, which takes a count in number of objects and also performs initialization and destruction.

Reply:
I like the latest iteration a lot more too. I think it is less error prone.